verifiers/internal/gha/verifier.go

```diff
@@ -80,11 +72,72 @@ func (v *GHAVerifier) VerifyArtifact(ctx context.Context,
 	return r, builderID, err
 }
 
+// VerifyArtifact verifies provenance for an artifact.
+func (v *GHAVerifier) VerifyArtifact(ctx context.Context,
+	provenance []byte, artifactHash string,
+	provenanceOpts *options.ProvenanceOpts,
+	builderOpts *options.BuilderOpts,
+) ([]byte, string, error) {
+	rClient, err := rekor.NewClient(defaultRekorAddr)
+	if err != nil {
+		return nil, "", err
+	}
+
+	/* Verify signature on the intoto attestation. */
+	env, cert, err := VerifyProvenanceSignature(ctx, rClient, provenance, artifactHash)
+	if err != nil {
+		return nil, "", err
+	}
+
+	return verifyEnvAndCert(env, cert, provenanceOpts, builderOpts)
+}
+
 // VerifyImage verifies provenance for an OCI image.
 func (v *GHAVerifier) VerifyImage(ctx context.Context,
-	provenance []byte, artifactHash string,
+	artifactReference string,
 	provenanceOpts *options.ProvenanceOpts,
 	builderOpts *options.BuilderOpts,
 ) ([]byte, string, error) {
-	return nil, "todo", serrors.ErrorNotSupported
+	/* Retrieve any valid signed attestations that chain up to Fulcio root CA. */
+	roots, err := fulcio.GetRoots()
+	if err != nil {
+		return nil, "", err
+	}
+	opts := &cosign.CheckOpts{
+		RootCerts: roots,
+	}
+
+	atts, _, err := container.RunCosignImageVerification(ctx,
```

Comment:
I would move this function into `gha/` in container.go or image.go

Reply:
my rationale for putting container externally was that it may be used across verifiers. WDYT about putting it in verifiers/container/?

Reply:
you convinced me sharing is beneficial in another comment already :p. Under `verifiers/container/cosign.go` SGTM.